clusters/resource_cluster.go

```diff
@@ -102,6 +102,17 @@ func SparkConfDiffSuppressFunc(k, old, new string, d *schema.ResourceData) bool
 	return false
 }
 
+// These are aliases which is the reason why we suppress these changes
+func DataSecurityModeDiffSuppressFunc(k, old, new string, d *schema.ResourceData) bool {
+	if old == "SINGLE_USER" && new == "DATA_SECURITY_MODE_DEDICATED" ||
+		old == "USER_ISOLATION" && new == "DATA_SECURITY_MODE_STANDARD" ||
+		(old == "SINGLE_USER" || old == "USER_ISOLATION") && new == "DATA_SECURITY_MODE_AUTO" {
+		log.Printf("[DEBUG] Suppressing diff for k=%#v old=%#v new=%#v", k, old, new)
```

Comment:
You can make it a bit simpler: 

```
{
  log.Printf("[DEBUG] Suppressing diff for k=%#v old=%#v new=%#v", k, old, new)
  return (old == "SINGLE_USER" && new == "DATA_SECURITY_MODE_DEDICATED") ||
	(old == "USER_ISOLATION" && new == "DATA_SECURITY_MODE_STANDARD") ||
	((old == "SINGLE_USER" || old == "USER_ISOLATION") && new == "DATA_SECURITY_MODE_AUTO")
}
```

and it's also better to put each OR alternative into brackets